examples/timeseries_classification_from_scratch.qmd

```diff
@@ -0,0 +1,173 @@
+---
+title: "Timeseries classification from scratch"
+authors:
+  - "[hfawaz](https://github.com/hfawaz/)"
+  - "[terrytangyuan](https://github.com/terrytangyuan) - R adaptation"
+  - "[t-kalinowski](https://github.com/t-kalinowski) - R adaptation"
+date-created: 2022/12/03
+date-last-modified: 2022/12/03
+description: "Training a timeseries classifier from scratch on the FordA dataset from the UCR/UEA archive."
+categories: [timeseries]
+aliases:
+  - ../guide/keras/examples/timeseries_classification_from_scratch/index.html
+---
+
+## Introduction
+
+This example shows how to do timeseries classification from scratch, starting from raw
+CSV timeseries files on disk. We demonstrate the workflow on the FordA dataset from the
+[UCR/UEA archive](https://www.cs.ucr.edu/%7Eeamonn/time_series_data_2018/).
+
+## Setup
+
+```{r}
+library(tensorflow)
+library(keras)
+set.seed(1234)
+```
+
+```{r, message=FALSE}
+train_df <- readr::read_tsv("https://raw.githubusercontent.com/hfawaz/cd-diagram/master/FordA/FordA_TRAIN.tsv", col_names = FALSE)
+x_train <- train_df[, -1]
+y_train <- train_df[, 1]
+
+test_df <- readr::read_tsv("https://raw.githubusercontent.com/hfawaz/cd-diagram/master/FordA/FordA_TEST.tsv", col_names = FALSE)
+x_test <- test_df[, -1]
+y_test <- test_df[, 1]
+```
+
+## Visualize the data
+
+Here we visualize one timeseries example for each class in the dataset.
+
+```{r}
+library(ggfortify)
```

Comment:
I hope this is okay. I used this package I maintain: https://github.com/sinhrks/ggfortify

It's easier to read, and readers don't have to learn ggplot2 syntax, which might be out of scope of this example.

Reply:
Looks nice! I think it's fine.